test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/endpointproviders/EndpointInterceptorTests.java

```diff
@@ -329,7 +332,135 @@ public void async_endpointProviderReturnsSignerProperties_overridesV4AAuthScheme
         assertThat(signer.request.property(AwsV4aHttpSigner.DOUBLE_URL_ENCODE)).isEqualTo(false);
     }
 
-    private static AuthScheme<?> capturingAuthScheme(String schemeId, CapturingSigner signer) {
+    @Test
+    public void sync_endpointProviderDoesNotReturnV4SignerProperties_executionAttributesFromAuthSchemeOption() {
+        RestJsonEndpointProvidersEndpointProvider defaultEndpointProvider =
+            RestJsonEndpointProvidersEndpointProvider.defaultProvider();
+        CapturingInterceptor interceptor = new CapturingInterceptor();
+
+        RestJsonEndpointProvidersAsyncClient client = asyncClientBuilder()
+            .overrideConfiguration(c -> c.addExecutionInterceptor(interceptor))
+            .authSchemeProvider(p -> singletonList(AuthSchemeOption.builder()
+                                                                   .schemeId("aws.auth#sigv4")
+                                                                   .putSignerProperty(AwsV4HttpSigner.REGION_NAME, "X")
+                                                                   .putSignerProperty(AwsV4HttpSigner.SERVICE_SIGNING_NAME, "Y")
+                                                                   .putSignerProperty(AwsV4HttpSigner.DOUBLE_URL_ENCODE, true)
+                                                                   .build()))
+            .endpointProvider(r -> defaultEndpointProvider.resolveEndpoint(r)
+                                                          .thenApply(e -> e.toBuilder()
+                                                                           .putAttribute(AwsEndpointAttribute.AUTH_SCHEMES, emptyList())
+                                                                           .build()))
+            .build();
+
+        assertThatThrownBy(() -> client.operationWithHostPrefix(r -> {}).join())
+            .hasMessageContaining("stop");
+
+        assertThat(interceptor.executionAttributes.getAttribute(AwsSignerExecutionAttribute.SIGNING_REGION)).isEqualTo(Region.of("X"));
+        assertThat(interceptor.executionAttributes.getAttribute(AwsSignerExecutionAttribute.SERVICE_SIGNING_NAME)).isEqualTo("Y");
+        assertThat(interceptor.executionAttributes.getAttribute(AwsSignerExecutionAttribute.SIGNER_DOUBLE_URL_ENCODE)).isEqualTo(true);
```

Comment:
Related to my comment above about `copySignerPropertiesToAttributes`, this is surprising behavior for Endpoint Interceptor. In this test case, Endpoint rules isn't playing a role in auth scheme/properties, but is still converting _some_ (just these 3) signer properties to execution attributes. If in this test case, a 4th SignerProperty was set in AuthSchemeOption the corresponding execution attribute would not be set and that is odd behavior.

Reply:
See my comment above for the ownership concerns. I agree that it's weird here and we should move it. To where? Let's talk on my reply above.

As for the new 4th signer property concern, I don't see that being a problem. If we added new signer properties, we probably won't bother adapting it to the old execution attributes - I can't think of a reason to do so. If that's right it  means the mapping code probably will never change to add support for new attributes.

Reply:
Yes, from a maintainability perspective it's confusing and error prone. I'd expect the attributes that are related to AuthSchemeOption to be set after auth scheme resolution, but there are maybe some blockers I don't know about.

Reply:
Clarified offline that the 4th signer property case I was thinking was not a new Signer property we'll add later, but other existing signer properties that map to existing execution attributes, like payload signing.